edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Metric.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class Metric
+    {
+        public DateTime TimeGeneratedUtc { get; }
+        public string Name { get; }
+        public double Value { get; }
+        public string Tags { get; }
+
+        public Metric(DateTime timeGeneratedUtc, string name, double value, string tags)
+        {
+            this.TimeGeneratedUtc = Preconditions.CheckNotNull(timeGeneratedUtc, nameof(timeGeneratedUtc));
+            this.Name = Preconditions.CheckNotNull(name, nameof(name));
+            this.Value = value;
+            this.Tags = Preconditions.CheckNotNull(tags, nameof(tags));
+        }
+
+        /// <summary>
+        /// This combines the hashes of name and tags to make it easy to group all metrics.
+        /// </summary>
+        /// <returns>Hash of name and tags.</returns>
+        public int HashNameAndTag()
```

Comment:
Try to avoid name having implementation details; what if later than a new field introduced and used in this method, then you need to update this method name.
```suggestion
        public int GetMetricKey()
```